pyomo/contrib/mindtpy/tests/test_mindtpy_lp_nlp.py

```diff
@@ -0,0 +1,155 @@
+"""Tests for the MINDT solver plugin."""
+from math import fabs
+import pyomo.core.base.symbolic
+import pyutilib.th as unittest
+from pyomo.contrib.mindtpy.tests.eight_process_problem import \
+    EightProcessFlowsheet
+from pyomo.contrib.mindtpy.tests.MINLP_simple import SimpleMINLP as SimpleMINLP
+from pyomo.contrib.mindtpy.tests.MINLP2_simple import SimpleMINLP as SimpleMINLP2
+from pyomo.contrib.mindtpy.tests.MINLP3_simple import SimpleMINLP as SimpleMINLP3
+from pyomo.contrib.mindtpy.tests.from_proposal import ProposalModel
+from pyomo.contrib.mindtpy.tests.online_doc_example import OnlineDocExample
+from pyomo.environ import SolverFactory, value
+
+required_solvers = ('ipopt', 'cplex_persistent')
+required_solvers_temp = ('ipopt', 'cplex')
+if all(SolverFactory(s).available() for s in required_solvers_temp):
+    subsolvers_available = True
+else:
+    subsolvers_available = False
+
+
+@unittest.skipIf(not subsolvers_available,
+                 "Required subsolvers %s are not available"
+                 % (required_solvers,))
+@unittest.skipIf(not pyomo.core.base.symbolic.differentiate_available,
+                 "Symbolic differentiation is not available")
+class TestMindtPy(unittest.TestCase):
+    """Tests for the MindtPy solver plugin."""
+
+    # lazy callback tests
+
+    def test_lazy_OA_8PP(self):
+        """Test the outer approximation decomposition algorithm."""
+        with SolverFactory('mindtpy') as opt:
+            model = EightProcessFlowsheet()
+            print('\n Solving problem with Outer Approximation')
+            opt.solve(model, strategy='OA',
+                      init_strategy='rNLP',
+                      mip_solver=required_solvers[1],
+                      nlp_solver=required_solvers[0],
+                      bound_tolerance=1E-5,
+                      single_tree=True)
+
+            # self.assertIs(results.solver.termination_condition,
+            #               TerminationCondition.optimal)
```

Comment:
Why do we not need to check for this anymore?

Reply:
Sure, we need to check the termination condition. I have added them back.